File: backend/app/schemas/menu.py
from decimal import Decimal
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class PublicMenuDish(BaseModel):
    id: UUID
    name: str
    description: str | None
    price: Decimal
    price_offer: Decimal | None
    image_url: str | None
    featured: bool
    tags: list[str]

    model_config = ConfigDict(from_attributes=True)


class PublicMenuCategory(BaseModel):
    id: UUID
    name: str
    description: str | None
    position: int
    dishes: list[PublicMenuDish]


class PublicMenuRestaurant(BaseModel):
    id: UUID
    name: str
    slug: str
    description: str | None
    logo_url: str | None
    phone: str | None
    address: str | None
    hours: dict | None


class PublicMenuResponse(BaseModel):
    restaurant: PublicMenuRestaurant
    categories: list[PublicMenuCategory]

File: frontend/src/types/menu.ts
export type PublicMenuDish = {
  id: string
  name: string
  description: string | null
  price: number | string
  price_offer: number | string | null
  image_url: string | null
  featured: boolean
  tags: string[]
}

export type PublicMenuCategory = {
  id: string
  name: string
  description: string | null
  position: number
  dishes: PublicMenuDish[]
}

export type PublicMenuRestaurant = {
  id: string
  name: string
  slug: string
  description: string | null
  logo_url: string | null
  phone: string | null
  address: string | null
  hours: Record<string, unknown> | null
}

export type PublicMenuResponse = {
  restaurant: PublicMenuRestaurant
  categories: PublicMenuCategory[]
}
